src/NServiceBus.Core.Tests/Sagas/SagasFeatureTest.cs

```diff
@@ -0,0 +1,92 @@
+﻿namespace NServiceBus.Core.Tests.Sagas
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq.Expressions;
+    using NServiceBus.Features;
+    using NServiceBus.Persistence;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Sagas;
+    using NUnit.Framework;
+    using ObjectBuilder;
+    using Settings;
+
+    [TestFixture]
+    public class SagasFeatureTest
+    {
+        [Test]
+        public void Throws_when_no_storage_for_sagas_selected()
+        {
+            var feature = new Sagas();
+            var settings = new SettingsHolder();
+            settings.Set("ResultingSupportedStorages", new List<Type>());
+
+            Assert.That(() => { feature.SetupFeature(new FeatureConfigurationContext(settings, null, null)); }, Throws.Exception.With.Message.Contains("DisableFeature<Sagas>()"));
+        }
+
+        [Test]
+        public void Doesnt_throw_when_storage_for_sagas_selected()
+        {
+            var feature = new Sagas();
+            var settings = new SettingsHolder();
+            settings.Set("ResultingSupportedStorages", new List<Type> { typeof(StorageType.Sagas) });
+            settings.Set<SagaMetadataCollection>(new SagaMetadataCollection());
+            settings.Set("TypesToScan", new List<Type>());
+
+            Assert.That(() => { feature.SetupFeature(new FeatureConfigurationContext(settings, new FakeConfigurer(), new PipelineSettings(new PipelineModifications()))); }, Throws.Nothing);
+        }
+
+        class FakeConfigurer : IConfigureComponents
```

Comment:
Would it work to use the funcbuilder here instead https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core.Tests/FuncBuilder.cs ?